internal/kafka/internal/services/cluster_placement_strategy.go

```diff
@@ -22,17 +24,79 @@ func NewClusterPlacementStrategy(clusterService ClusterService, dataplaneCluster
 	case dataplaneClusterConfig.IsDataPlaneAutoScalingEnabled():
 		clusterSelection = &FirstReadyWithCapacity{clusterService, kafkaConfig}
 	default:
-		clusterSelection = &FirstReadyCluster{clusterService}
+		clusterSelection = &FirstReadyCluster{clusterService, kafkaConfig}
 	}
 	return clusterSelection
 }
 
+// findDesiredDataPlaneClusterIfItHasCapacityAvailable finds and returns the desired data plane cluster if
+// if it has capacity available.
+type findDesiredDataPlaneClusterIfItHasCapacityAvailable struct {
```

Comment:
Is this the name of the placement strategy? Would something like `findDataPlaneClusterByIdIfItHasCapacityAvailable` be more clear?

Reply:
Yes, that's the name of the placement strategy. 
Notice how it has been called from the other placement strategy as well.

I am okay with changing the name to what you are suggesting.

Reply:
done